apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/Inscriptions/useInscriptionsModel.tsx

```diff
@@ -5,21 +5,24 @@ import { InscriptionsItemProps } from "LLD/features/Collectibles/types/Inscripti
 
 type Props = {
   inscriptions: SimpleHashNft[];
+  onInscriptionClick: (inscription: SimpleHashNft) => void;
 };
 
-export const useInscriptionsModel = ({ inscriptions }: Props) => {
+export const useInscriptionsModel = ({ inscriptions, onInscriptionClick }: Props) => {
   const [displayShowMore, setDisplayShowMore] = useState(false);
   const [displayedObjects, setDisplayedObjects] = useState<InscriptionsItemProps[]>([]);
 
   const items: InscriptionsItemProps[] = useMemo(
-    () => getInscriptionsData(inscriptions),
-    [inscriptions],
+    () => getInscriptionsData(inscriptions, onInscriptionClick),
+    [inscriptions, onInscriptionClick],
   );
 
   useEffect(() => {
-    if (items.length > 3) setDisplayShowMore(true);
-    setDisplayedObjects(items.slice(0, 3));
-  }, [items]);
+    if (displayedObjects.length === 0) {
```

Comment:
I don't understand this

Reply:
It's to avoid to trigger it at each re render. Sometimes (and when a drawer just opened) displayedObjects comes back to his initial state with only 3 object in it. So I trigger it only at the first render

Reply:
I've changed to not do it with a use Effect to make easier to read